pilot/pkg/model/push_context.go

```diff
@@ -311,20 +317,36 @@ func (ps *PushContext) UpdateMetrics() {
 
 // Services returns the list of services that are visible to a Proxy in a given config namespace
 func (ps *PushContext) Services(proxy *Proxy) []*Service {
-	// TODO: use network scopes here, and return services in the proxy namespace if
-	// configured to use current_namespace as the scope and proxy is not nil
-	//if proxy == nil {
-	//	return ps.allServices
-	//}
-	return ps.allServices
+	out := []*Service{}
+	out = append(out, ps.publicServices...)
+
+	if proxy == nil {
+		for _, privateServices := range ps.privateServicesByNamespace {
+			out = append(out, privateServices...)
+		}
+	} else {
+		out = append(out, ps.privateServicesByNamespace[proxy.ConfigNamespace]...)
+	}
```

Comment:
can you invert the order here? Add private services first and then add the public services